### PR TITLE
Fix chunk sharding

### PR DESCRIPTION
**Pull request by mckinsel**
The `chunk_reads` steps are supposed to break up big fastq input files into little ones, using the `reads_per_file` parameter. But previously, if it actually did that, the pipeline would break. This change fixes the way that the chunked fastqs are distributed to subsequent workflow tasks.

### CODE REVIEW

File: 10x/count/count.wdl [hunk 1]
```diff
@@ -137,6 +137,11 @@ task chunk_reads_main {
     
     # Read the out_chunks definitions into a file
     jq '.out_chunks' _outs > out_chunks.json
+    jq -r  'map(.read_chunks | to_entries | map(.value)) | map(join(" ")) | .[]' out_chunks.json > chunk_filenames.lines
```
Comment (ambrosejcarr) [inline 10x/count/count.wdl]: Sorry I still need a bit of hand-holding for code blocks like this.

File: 10x/count/count.wdl [hunk 2]
```diff
@@ -173,6 +182,17 @@ task chunk_reads_join {
     
     # Write out_chunks to their own file
     jq '.out_chunks' _outs > out_chunks.json
+
```
Comment (ambrosejcarr) [inline 10x/count/count.wdl]: ... and also some explanation here.

File: 10x/count/count.wdl [hunk 3]
```diff
@@ -161,7 +161,7 @@ task chunk_reads_join {
     
     # Create the args
     echo "{}" > _args
-    ehco "{}" > _chunk_defs
+    echo "{}" > _chunk_defs
```
Comment (ambrosejcarr) [inline 10x/count/count.wdl]: I'm sad to see `ehco` go.

File: 10x/count/count.wdl (hunk 1 above)
Comment (mckinsel) [reply 10x/count/count.wdl]: No problem, added some explanation. I'm basically just tarring the fastq chunks, but I have to parse the JSON file to know which files to include in which tarball.

File: 10x/count/count.wdl (hunk 2 above)
Comment (mckinsel) [reply 10x/count/count.wdl]: This is a very sad piece of code, but I added some comments. All I want to do is flatten a list of lists of files, but so far I seem to have to transfer everything to worker to do that.

File: 10x/count/count.wdl [hunk 4]
```diff
@@ -148,20 +172,24 @@ task chunk_reads_main {
...
-    Array[File] fastq_chunks = glob("files/*.fastq")
+    Array[File] fastq_chunks = glob("fastq_chunk_*.tar.gz")
   }
 }
 
 task chunk_reads_join {
   # Array of outs objects from the chunk_reads_main steps. All this step is doing
   # is joining some dicts together.
   Array[File] outs
+  Array[Array[File]] fastq_chunks
```
Comment (ambrosejcarr) [inline 10x/count/count.wdl]: I think for maintainability we should put some comments around inputs that aren't clear from their name/type that describe what they are. What you've got on lines 149 + 150 is good; we could copy it here. What do you think?

File: 10x/count/count.wdl (hunk 1 above)
Comment (samanehsan) [reply 10x/count/count.wdl]: I thought your explanation was really helpful! Thanks for adding it in @mckinsel

File: 10x/count/count.wdl (hunk 4 above)
Comment (mckinsel) [reply 10x/count/count.wdl]: Yeah good idea. I added a little note about what the `fastq_chunks` are supposed to be.

**Pull request merged**